echo-telemetry/src/test/kotlin/com/netflix/spinnaker/echo/telemetry/InstanceIdSupplierTest.kt

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2020 Google, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License")
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.echo.telemetry
+
+import com.netflix.spinnaker.echo.config.TelemetryConfig.TelemetryConfigProps
+import com.netflix.spinnaker.kork.jedis.EmbeddedRedis
+import com.netflix.spinnaker.kork.jedis.JedisClientDelegate
+import com.netflix.spinnaker.kork.jedis.RedisClientSelector
+import org.junit.jupiter.api.Test
+import strikt.api.expectThat
+import strikt.assertions.isEqualTo
+
+class InstanceIdSupplierTest {
```

Comment:
Is there a way to mock some part of Redis so that you can have `c.setnx` throw an exception?

Reply:
I started looking into that, and then felt like I was basically testing the implementation of our redis autowiring (and that the test would mostly catch any changes to how we do that). But given my catch statement was incorrect and I didn't catch it, let me see if I can come up with a reasonable test.

Reply:
I think there are other reasons it could throw an exception like a network issue or something. But yeah, I dunno, if it's a ton of work, it's not really that big of a deal.

Reply:
I was able to reasonably mock a `JedisException` so now have a test that ensures we catch these appropriately.